### PR TITLE
feat(scaffolder-backend-module-github): add github:branch-protection:create action

### PR DESCRIPTION
**Pull request by koalaty-code**
## Hey, I just made a Pull Request!

This adds a new Scaffolder Action to create/update GitHub Branch Protection settings for a repository branch.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by camilaibs** [comment]
Hi @koalaty-code 👋🏻 , could you please address the build errors?
For example: https://github.com/backstage/backstage/actions/runs/11077393144/job/30782602273?pr=26886

**Review by camilaibs — APPROVED** [review]
Looks great to me, left only minor suggestions on the changeset file 🙂 . I'd appreciate a review from @backstage/scaffolder-maintainers and I'd like to double check with @backstage/maintainers if it is necessary to add you as owner of this module in the CODEOWNERS file.

File: plugins/scaffolder-backend-module-github/src/actions/githubBranchProtection.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    requiredStatusCheckContexts?: string[];
+    requireBranchesToBeUpToDate?: boolean;
+    requiredConversationResolution?: boolean;
+    requireLastPushApproval?: boolean;
+    requiredCommitSigning?: boolean;
+    token?: string;
+  }>({
+    id: 'github:branch-protection:create',
+    description: 'Configures Branch Protection',
+    examples: undefined, // TODO
```
[the remark this answers is not in the document]
Comment (koalaty-code) [reply plugins/scaffolder-backend-module-github/src/actions/githubBranchProtection.ts]: Updated. I forgot to set the examples here after creating them.

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Thanks for this! :praya:

**Pull request merged**